pennylane/devices/default_clifford.py

```diff
@@ -0,0 +1,712 @@
+    ) -> Result_or_ResultBatch:
+        is_single_circuit = False
+        if isinstance(circuits, QuantumScript):
+            is_single_circuit = True
+            circuits = [circuits]
+
+        max_workers = execution_config.device_options.get("max_workers", self._max_workers)
+        interface = (
+            execution_config.interface
+            if execution_config.gradient_method in {"backprop", None}
+            else None
+        )
+        if max_workers is None:
+            results = tuple(
+                self.simulate(
+                    c,
+                    rng=self._rng,
+                    prng_key=self._prng_key,
+                    debugger=self._debugger,
+                    interface=interface,
+                )
+                for c in circuits
+            )
+        else:
+            vanilla_circuits = [convert_to_numpy_parameters(c) for c in circuits]
+            seeds = self._rng.integers(2**31 - 1, size=len(vanilla_circuits))
+            _wrap_simulate = partial(self.simulate, debugger=None, interface=interface)
+            with concurrent.futures.ProcessPoolExecutor(max_workers=max_workers) as executor:
+                exec_map = executor.map(
+                    _wrap_simulate,
+                    vanilla_circuits,
+                    seeds,
+                    [self._prng_key] * len(vanilla_circuits),
+                )
+                results = tuple(exec_map)
+
+            # reset _rng to mimic serial behavior
+            self._rng = np.random.default_rng(self._rng.integers(2**31 - 1))
+
+        if self.tracker.active:
+            self.tracker.update(batches=1)
+            self.tracker.record()
+            for i, c in enumerate(circuits):
+                qpu_executions, shots = get_num_shots_and_executions(c)
+                res = np.array(results[i]) if isinstance(results[i], Number) else results[i]
+                if c.shots:  # pragma: no cover
+                    self.tracker.update(
+                        simulations=1,
+                        executions=qpu_executions,
+                        results=res,
+                        shots=shots,
+                        resources=c.specs["resources"],
+                    )
+                else:
+                    self.tracker.update(
+                        simulations=1,
+                        executions=qpu_executions,
+                        results=res,
+                        resources=c.specs["resources"],
+                    )
+                self.tracker.record()
+
+        return results[0] if is_single_circuit else results
+
+    def supports_derivatives(
+        self,
+        execution_config: Optional[ExecutionConfig] = None,
+        circuit: Optional[QuantumTape] = None,
+    ) -> bool:
+        """Check whether or not derivatives are available for a given configuration and circuit.
+
+        ``DefaultClifford`` returns trivial derivates everytime.
```

Comment:
if the derivatives are trivial, why not just set this to return False, and remove everything else related to derivatives?

Reply:
It is trivial for now, but we might need them when we add the `T` support.

Reply:
in this case, I would prefer we remove a lot of the speculative code from this file, primarily anything related to the execution config. we can always copy-paste from default.qubit later if we need it : )

Reply:
Just want to confirm if @timmysilv's comment was resolved? I would also like to keep it simple for now and not make any future assumptions about T-gate support.

Reply:
Yes! We are not making any complex assumptions for T-gate support.